satpy/readers/generic_image.py

```diff
@@ -102,19 +106,50 @@ def end_time(self):
 
     def get_dataset(self, key, info):
         """Get a dataset from the file."""
-        logger.debug("Reading %s.", key)
-        return self.file_content[key['name']]
-
-
-def mask_image_data(data):
-    """Mask image data if alpha channel is present."""
-    if data.bands.size in (2, 4):
-        if not np.issubdtype(data.dtype, np.integer):
-            raise ValueError("Only integer datatypes can be used as a mask.")
-        mask = data.data[-1, :, :] == np.iinfo(data.dtype).min
-        data = data.astype(np.float64)
-        masked_data = da.stack([da.where(mask, np.nan, data.data[i, :, :])
-                                for i in range(data.shape[0])])
-        data.data = masked_data
-        data = data.sel(bands=BANDS[data.bands.size - 1])
-    return data
+        ds_name = self.dataset_name if self.dataset_name else key['name']
+        logger.debug("Reading '%s.'", ds_name)
+        data = self.file_content[ds_name]
+
+        # Mask data if necessary
+        try:
+            data = self.__mask_image_data(data, info)
+        except ValueError as err:
+            logger.warning(err)
+
+        return data
+
+    def __mask_image_data(self, data, info):
```

Comment:
Why did this become a class method? Couldn't/shouldn't this remain a function outside the class? Also, please use a single underscore. Double underscore has special handling in python and isn't needed here.

Reply:
Sorry, too much java in the last months. I'll fix it tomorrow morning.

Reply:
Fixed it.